### PR TITLE
feat: Description for cluster and apps

### LINKED ISSUE #4168: Feature: Add short description for apps and clusters
### 🔖 Feature description

Add short description for devtron apps, helm apps, jobs and clusters

### 🎤 Pitch

Currently devtron only have option for full fledged readme but there is no option for having a brief info about the app and its working. 

### ✌️ Solution

Introduce new description field and rectify/rename older ones and notes/readme. 

### 🔄️ Alternative

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

AB#6475

### PR DESCRIPTION
**Pull request by kartik-579**
# Description

Fixes #4168

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Review by vikramdevtron — DISMISSED** [review]

File: api/restHandler/CoreAppRestHandler.go [hunk 1]
```diff
@@ -2378,7 +2378,7 @@ func (handler CoreAppRestHandlerImpl) GetAppWorkflowAndOverridesSample(w http.Re
 		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
 		return
 	}
-	app, err := handler.appCrudOperationService.GetAppMetaInfo(appId)
+	app, err := handler.appCrudOperationService.GetAppMetaInfo(appId, 0, 0)
```
Comment (vikramdevtron) [inline api/restHandler/CoreAppRestHandler.go:2381]: We can pass variable name indicating specific ID. and that can be for all the occurrence.

File: pkg/app/AppCrudOperationService.go [hunk 2]
```diff
@@ -284,7 +292,7 @@ func (impl AppCrudOperationServiceImpl) FindAll() ([]*bean.AppLabelDto, error) {
 	return results, nil
 }
 
-func (impl AppCrudOperationServiceImpl) GetAppMetaInfo(appId int) (*bean.AppMetaInfoDto, error) {
+func (impl AppCrudOperationServiceImpl) GetAppMetaInfo(appId int, installedAppId int, installedAppEnvId int) (*bean.AppMetaInfoDto, error) {
```
Comment (vikramdevtron) [inline pkg/app/AppCrudOperationService.go:295]: installedAppEnvId is this taking environment id if yes, rename it to environmentId

File: pkg/pipeline/BuildPipelineConfigService.go [hunk 3]
```diff
@@ -440,8 +440,8 @@ func (impl *CiPipelineConfigServiceImpl) buildPayloadOption() []bean.PayloadOpti
 func (impl *CiPipelineConfigServiceImpl) buildExternalCiWebhookSchema() map[string]interface{} {
 	schema := make(map[string]interface{})
 	schema["dockerImage"] = &bean.SchemaObject{Description: "docker image created for your application (Eg. quay.io/devtron/test:da3ba325-161-467)", DataType: "String", Example: "test-docker-repo/test:b150cc81-5-20", Optional: false}
```
Comment (vikramdevtron) [inline pkg/pipeline/BuildPipelineConfigService.go:442]: is this sample required here ?

File: pkg/resourceGroup/ResourceGroupService.go [hunk 4]
```diff
@@ -132,7 +132,7 @@ type ResourceGroupDto struct {
 //	AppId           int    `json:"appId,omitempty"`
 //	AppName         string `json:"appName,omitempty"`
```
Comment (vikramdevtron) [inline pkg/resourceGroup/ResourceGroupService.go:133]: remove unwanted commented lines

File: api/restHandler/CoreAppRestHandler.go (hunk 1 above)
Comment (kartik-579) [reply api/restHandler/CoreAppRestHandler.go:2381]: updated

File: pkg/app/AppCrudOperationService.go (hunk 2 above)
Comment (kartik-579) [reply pkg/app/AppCrudOperationService.go:295]: changed with added comment for easier understanding

File: pkg/pipeline/BuildPipelineConfigService.go (hunk 3 above)
Comment (kartik-579) [reply pkg/pipeline/BuildPipelineConfigService.go:442]: removed

File: pkg/resourceGroup/ResourceGroupService.go (hunk 4 above)
Comment (kartik-579) [reply pkg/resourceGroup/ResourceGroupService.go:133]: removed

**Review by vikramdevtron — APPROVED** [review]

**Review by vikramdevtron — APPROVED** [review]

**Pull request merged**